### ISSUE TITLE
OpenIDConnectAuth discover is missing encryption ciphers from RFC7518

### ISSUE THREAD

**Issue by gofabian** [eclipse-vertx/vertx-auth#279]
### Version

* vert.x core: 3.7.0
* vert.x auth: 3.7.0

### Context

When discovering OpenID Connect options from `<url>/.well-known/openid-configuration` Vert.x will read the `alg` attribute from the `jwks_uri`. Here Vert.x only supports the signing ciphers for JWS:

https://tools.ietf.org/html/rfc7518#appendix-A.1
HS256
HS384
HS512
RS256
RS384
RS512
ES256
ES384
ES512
PS256
PS384
PS512

but not the encryption ciphers for JWE:

https://tools.ietf.org/html/rfc7518#appendix-A.2
RSA1_5
RSA-OAEP
RSA-OAEP-256
ECDH-ES
A128KW
A192KW
A256KW

### Do you have a reproducer?

In my case the `jwks_uri` provides something like this:

```
{
  "keys": [
    {
      "kty": "RSA",
      "kid": "...",
      "use": "enc",
      "alg": "RSA-OAEP",
      "n": "...",
      "e": "..."
    },
    {
      "kty": "RSA",
      "kid": "...",
      "use": "sig",
      "alg": "RS256",
      "n": "...",
      "e": "..."
    },
...
}
```

`OpenIDConnectAuth.discover(...)` results in a failed future:

```
java.lang.RuntimeException: java.security.NoSuchAlgorithmException: RSA-OAEP
	at io.vertx.ext.jwt.JWK.<init>(JWK.java:186)
	at io.vertx.ext.auth.oauth2.impl.OAuth2AuthProviderImpl.lambda$loadJWK$0(OAuth2AuthProviderImpl.java:145)
	at io.vertx.ext.auth.oauth2.impl.OAuth2API.lambda$null$1(OAuth2API.java:129)
	at io.vertx.core.http.impl.HttpClientResponseImpl$BodyHandler.notifyHandler(HttpClientResponseImpl.java:296)
	at ...
Caused by: java.security.NoSuchAlgorithmException: RSA-OAEP
	at io.vertx.ext.jwt.JWK.createRSA(JWK.java:202)
	at io.vertx.ext.jwt.JWK.<init>(JWK.java:173)
	... 46 more
```

### Suggestion

I suggest we should not fail the discovery but just ignore the unknown algorithms. An alternative is to support all ciphers from RFC7518. 

What do you think?

**Comment by pmlopes** [comment]
@gofabian yes i agree we should not fail and give a warning that the missing alg will not be available.

**Comment by pmlopes** [comment]
Fixed by #280
